addons/stock_account/migrations/8.0.1.1/post-migrate.py

```diff
@@ -42,7 +44,65 @@ def create_properties(cr, pool):
                                context=ctx)
 
 
-@openupgrade.migrate()
+def propagate_invoice_state(cr):
+    """ Invoice state is now propagated from sale order to procurement to
+    stock move and picking. We trace it back from the picking and update
+    stock moves and procurements. """
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE stock_move sm
```

Comment:
Yeah, seems curious. Reading about that, PostgreSQL use the same optimizer for this clause than with INNER JOIN, so it should be in a good shape. Have you tried with a large set of data?

Anyways, please put a comment about what you told me for others that come to this code.